uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -151,11 +151,17 @@ function tasksassignments_civicrm_alterSettingsFolders(&$metaDataFolders = NULL)
   _tasksassignments_civix_civicrm_alterSettingsFolders($metaDataFolders);
 }
 
-function tasksassignments_civicrm_alterAPIPermissions($entity, $action, &$params, &$permissions)
-{
-    $permissions['contact']['get'] = array();
-    $permissions['contact']['getquick'] = array();
-  if ($entity == 'document' || $entity == 'task' || $entity == 'assignment') {
+function tasksassignments_civicrm_alterAPIPermissions($entity, $action, &$params, &$permissions) {
+  $permissions['contact']['get'] = array();
+  $permissions['contact']['getquick'] = array();
+  $entitiesToAvoidPermissions = [
```

Comment:
Unfortunately, yes. As I mentioned on the PR description, this is not a secure way to do this. I'm just following the existing approach in the extension. We need to fix a lot of things regarding T&A and API permissions, but that would be out of the scope of this ticket. I'm going to flag this to Mike and hopefully we'll have another ticket to handle this properly